### PR TITLE
Allows the use of a custom MarkupExtension with or without the suffix "Extension"

### LINKED ISSUE #2918: XAML markup extension requires full name
## Current behavior

UWP allows the developer to omit the "Extension" suffix when referring to XAML Markup extensions.

For example for:

```
public class IconFontExtension : MarkupExtension
```

We can use the following in XAML:

```
Icon={ex:IconFont ...}
```

In case of Uno, this currently causes a build error:

> System.InvalidOperationException: The type {using:TestApp.Shared.Xaml.MarkupExtensions}IconFont could not be found 

## Expected behavior

Omitting `Extension` suffix should be allowed.

## How to reproduce it (as minimally and precisely as possible)

1. Create a XAML markup extension with `Extension` suffix in its name.
2. Use it in XAML without the suffix.

## Environment

Nuget Package: 2.2 (dev)

Package Version(s): 2.2. (dev)

Affected platform(s):

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Windows
- [x] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by workgroupengineering**
GitHub Issue (If applicable): fixes #2918

## PR Type

What kind of change does this PR introduce?

- Bugfix
- Feature

## What is the current behavior?

Allows the use of a custom MarkupExtension only with fullname

## What is the new behavior?

Allows the use of a custom MarkupExtension with or without the suffix "Extension"
Example:

```csharp
        public class SimpleExtension : MarkupExtension
	{
		protected override object ProvideValue()
		{
			return null;
		}
	}
```

you can use:

```xaml
   <TextBlock Text="{l:SimpleExtension}"/>
```

or

```xaml
   <TextBlock Text="{l:Simple}"/>
```

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by davidjohnoliver — APPROVED** [review]
Thanks for the contribution!

You've left the PR in draft - if you're interested in putting the cherry on it, you could add a functional test in Uno.UI.Tests: https://github.com/unoplatform/uno/tree/master/src/Uno.UI.Tests/Windows_UI_Xaml/MarkupExtensionTests

But IMO it's mergeable with the test you've already added, which validates that the parsed Xaml doesn't fail the build.

File: src/Uno.UI.Tests/App/Xaml/Test_MarkupExtension.xaml [hunk 1]
```diff
@@ -47,6 +48,16 @@
 
 		<TextBlock x:Name="Text10"
 				   DataContext="{ex:NoReturnTypeMarkupExt UseValue=UseValue1, Value1={StaticResource TestEntity}}" />
+
+		<TextBlock x:Name="Text11"
+				   Text="{ex:SimpleMarkupExtExtesion TextValue='Just a simple ... fullname'}" />
```
Comment (jeromelaban) [inline src/Uno.UI.Tests/App/Xaml/Test_MarkupExtension.xaml:53]:
```suggestion
				   Text="{ex:SimpleMarkupExtExtension TextValue='Just a simple ... fullname'}" />
```

File: src/Uno.UI.Tests/App/Xaml/Test_MarkupExtension.xaml [hunk 2]
```diff
@@ -47,6 +48,16 @@
...
+
+		<TextBlock x:Name="Text12"
+				   Text="{ex:SimpleMarkupExtExtesion TextValue={StaticResource ResourceStringFull}}" />
```
Comment (jeromelaban) [inline src/Uno.UI.Tests/App/Xaml/Test_MarkupExtension.xaml:56]:
```suggestion
				   Text="{ex:SimpleMarkupExtExtension TextValue={StaticResource ResourceStringFull}}" />
```

File: src/Uno.UI.Tests/App/Xaml/Test_MarkupExtension.xaml [hunk 3]
```diff
@@ -47,6 +48,16 @@
...
+
+		<TextBlock x:Name="Text13"
+				   beh:AttachedPropertiesBehavior.CustomText="{ex:SimpleMarkupExt TextValue='String from attached property Fullname'}" />
```
Comment (jeromelaban) [inline src/Uno.UI.Tests/App/Xaml/Test_MarkupExtension.xaml:59]: Is this a typo or did you want to use `SimpleMarkup` instead ?

Comment (workgroupengineering) [reply src/Uno.UI.Tests/App/Xaml/Test_MarkupExtension.xaml:59]: is typo

File: src/Uno.UI.Tests/App/Xaml/Test_MarkupExtension.xaml (hunk 2 above)
Comment (workgroupengineering) [reply src/Uno.UI.Tests/App/Xaml/Test_MarkupExtension.xaml:56]: is typo

File: src/Uno.UI.Tests/App/Xaml/Test_MarkupExtension.xaml (hunk 1 above)
Comment (workgroupengineering) [reply src/Uno.UI.Tests/App/Xaml/Test_MarkupExtension.xaml:53]: is typo

**Comment by jeromelaban** [comment]
Looks like there's still an error there: 

https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=20970&view=logs&j=472d0e2c-70a8-549f-4788-02c7fe22e99c&t=6785d006-8ec7-5e10-add7-e59fa1ed348f&l=17956

One of the fields is not taken into account.

**Comment by workgroupengineering** [comment]
> 
> 
> Looks like there's still an error there:
> 
> https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=20970&view=logs&j=472d0e2c-70a8-549f-4788-02c7fe22e99c&t=6785d006-8ec7-5e10-add7-e59fa1ed348f&l=17956
> 
> One of the fields is not taken into account.

I'm working on it in my spare time. When I just fixed it, I will post the changes.

**Comment by jeromelaban** [comment]
> I'm working on it in my spare time. When I just fixed it, I will post the changes.

No worries at all, there's no implied rush, thanks for your contributions!

**Review by davidjohnoliver — APPROVED** [review]

**Pull request merged**